tools/ReleaseProcess/UploadArtifactsToGitHub.ps1

```diff
@@ -0,0 +1,444 @@
+## Copyright (c) Microsoft. All rights reserved.
+## Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+<#
+    .SYNOPSIS
+        This function downloads and prepares the Iot Edgelet artifacts from DevOps pipeline for a release. 
+        The final artifacts are collected under "<WorkDir>/output" path.
+    .NOTES
+        Requires Azure CLI to be login
+    .DESCRIPTION
+        1) Download edgelet artifacts from a given BuildId from a DevOps pipeline
+        2) Extract the *.zip artifacts into their own directories
+        3) Traverse the directories and rename a relavant artifact into the following format: 
+              .DEB _________________ {component}_{version}_{os}_{arch}.deb 
+              .RPM _________________ {component}-{version}.{os}.{arch}.rpm  (No name change)
+        4) Move the renamed artifacts to a staging directory at "<WorkDir>/output"
+    .PARAMETER BuildId
+        DevOps build ID for the downloading artifacts
+    .PARAMETER WorkDir
+        Absolute path of the working directory
+#>
+function Prepare-DevOps-Artifacts
+{
+    [CmdletBinding()]
+    param (
+        <# 
+        DevOps build ID for the downloading artifacts
+        #>
+        [Parameter(Mandatory)]
+        [string]
+        $BuildId,
+
+        <# 
+        Absolute path of the working directory
+        #>
+        [Parameter(Mandatory)]
+        [string]
+        $WorkDir
+    )
+
+    # Assume Az CLI is installed & logged in.
+    $pat = $(az keyvault secret show -n IotEdge1-PAT-msazure --vault-name edgebuildkv | ConvertFrom-Json)
```

Comment:
That's a great advice. I had an iffy feeling about using Az CLI but I didn't know how to get around it. Pushing the responsibility to the user is a good idea.

Reply:
I don't mind using the `az` CLI in the script, I just wanted to avoid the hard-coded vault+secret name. There are lots of ways to avoid that:
- _Use hard-coded secret names, but make the key vault configurable via a variable._ I believe the end-to-end tests do this
- _Use variables for secrets names and key vault name._ This results in too many variables IMO, and since secrets are already an abstraction over the underlying value, adding variables for secret names feels like too many levels of indirection...potentially confusing
- _Passing in secret values as variables._ More variables than the 1st option, but fewer variables than the 2nd option, and provides more flexibility (this is what I proposed above)
- I'm sure there are other options too...